docs/source/development/start/setup.rst

```diff
@@ -0,0 +1,165 @@
+During development, you may want to check how your changes affect the behavior of the package.
+LineaPy supports several interfaces to run on, and the relevant instructions are documented :ref:`here <interfaces>`.
+Please select the most convenient option for you.
+
+.. note::
+
+    When running LineaPy, make sure that the virtual environment has been activated.
+
+Development
+-----------
+
+Making Changes
+**************
+
+With development dependencies installed, you are now ready to contribute to the source code!
+First, make a separate branch for your development work. Please use an informative name so that
+others can get good sense of what your changes are about.
+
+.. code:: bash
+
+    git checkout -b <NEW-BRANCH-NAME>
+
+After making changes you desire, save them to your development branch:
+
+.. code:: bash
+
+    git add <PATH-TO-CHANGED-FILE>
+    git commit -m "<COMMIT-MESSAGE>"
+
+.. note::
+
+    To learn more about saving changes in Git, check this `tutorial <https://www.atlassian.com/git/tutorials/saving-changes>`_.
+
+Note that these changes have been saved only locally at this point, and you need to "push" them to your forked repo on GitHub:
+
+.. code:: bash
+
+    git push
+
+If the new (development) branch has not been pushed before, you will need to create its counterpart on GitHub with:
+
+.. code:: bash
+
+    git push --set-upstream origin <NEW-BRANCH-NAME>
+
+Testing Changes
+***************
+
+Testing is an important part of ``lineapy``'s development as it ensures that all features stay functional after changes.
+Hence, we strongly recommend you add tests for changes you introduce (check this :ref:`tutorial <add_test>` for adding tests).
+
+To run all tests (beware that this may take a while to complete):
+
+.. code:: bash
+
+    pytest tests
+
+Or, to run a particular test (e.g., one that you added/modified):
+
+.. code:: bash
+
+    pytest <PATH-TO-TEST-FILE>
+
+Integrating Changes
+*******************
+
+As you make your changes in your development branch, it is very possible that the original ``lineapy`` repo is updated by other developers.
+To ensure that your changes are compatible with these updates by others, you will need to regularly "sync" your development branch with the original
+``lineapy`` repo. You can do this by first syncing the ``main`` branch between your local (forked) repo and the original ``lineapy`` repo:
+
+.. code:: bash
+
+    git fetch upstream
+    git checkout main
+    git merge upstream/main
+
+Then, sync your development branch with the updated ``main`` branch:
+
+.. code:: bash
+
+    git checkout <DEV-BRANCH-NAME>
+    git rebase main
+
+.. note::
+
+    If updates in the original ``lineapy`` repo are not compatible with changes in your development branch,
+    you will need to resolve merge conflict(s). Check this
+    `tutorial <https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/addressing-merge-conflicts/resolving-a-merge-conflict-using-the-command-line>`_
```

Comment:
😭

Reply:
maybe have a comment here saying to reach out if they're stuck here? idk if we want do be "merge conflict hotline" but this could be a blocking point for newer contributors.

Reply:
Good point, will do.